glm/detail/type_vec2.inl

```diff
@@ -106,28 +106,60 @@ namespace glm
 	GLM_FUNC_QUALIFIER GLM_CONSTEXPR T & vec<2, T, Q>::operator[](typename vec<2, T, Q>::length_type i)
 	{
 		GLM_ASSERT_LENGTH(i, this->length());
-		switch(i)
+
+#		if GLM_LANG & GLM_LANG_CXX14_FLAG || GLM_LANG & GLM_LANG_CXX17_FLAG
+#		if GLM_LANG & GLM_LANG_CXX20_FLAG
+		if (std::is_constant_evaluated())
```

Comment:
I have to say the `#` not being close to the `if` is really making this hard to read

Reply:
i was trying to follow the already existing glm code style

Reply:
tho i agree